src/openforms/submissions/models/submission.py

```diff
@@ -256,6 +256,14 @@ class Submission(models.Model):
             "Indicates whether sensitive data (if there was any) has been removed from this submission."
         ),
     )
+    object_reference = models.CharField(
+        _("object reference"),
+        blank=True,
+        help_text=_(
+            "The object reference-id which will be used to modify the product in the registration backend. "
+            "(For example the object reference for the Object in the Objects API)"
+        ),
+    )
```

Comment:
```suggestion
    initial_data_reference = models.CharField(
        _("initial data reference"),
        blank=True,
        help_text=_(
            "An identifier that can be passed as a querystring when the form is started. Initial form field values are pre-populated from the retrieved data. During registration, the object may be updated again (or a new record may be created). This can be an object reference in the Objects API, for example."
        ),
    )
```

* I'd make the scope of this field clearer by naming it `initial_data_reference` - there will be other cases in the future where data is retrieved somewhere else from than the objects API (e.g. an existing zaak/case).
* generalized the help text and meaning - it's not guaranteed that we will update the same object again, you could also load data from the objects API but then write the registration to another place.

Reply:
Yes, makes sense I will update it. This means I should change it everywhere right?I mean serializers and SDK

Reply:
yep!